src/model/utils/selection-post-fixer.js

```diff
@@ -212,34 +227,45 @@ function tryFixingNonCollapsedRage( range, schema ) {
 	return null;
 }
 
-// Expands selection so it contains whole limit node.
+// Finds the outer-most ancestor.
 //
-// @param {module:engine/model/position~Position} position
+// @param {module:engine/model/node~Node} startingNode
 // @param {module:engine/model/schema~Schema} schema
 // @param {String} expandToDirection Direction of expansion - either 'start' or 'end' of the range.
-// @returns {module:engine/model/position~Position}
-function expandSelectionOnIsLimitNode( position, schema, expandToDirection ) {
-	let node = position.parent;
-	let parent = node;
+// @returns {module:engine/model/node~Node}
+function findOuterMostIsLimitAncestor( startingNode, schema ) {
```

Comment:
I'd call it `findOutermostLimitAncestor`. The `is` part makes the name confusing. I know that we are looking for an object that passes `isLimit()` in `Schema` but still. Also AFAIR outermost is a one word.